### PR TITLE
move arrow-comment of match arm inside the block

### LINKED ISSUE #3170: Bracing for match arm doesn't include comment
Before:
```rust
        Referrer::Client =>
            // it should be impossible for referrer to be anything else during fetching
            // https://fetch.spec.whatwg.org/#concept-request-referrer
           unreachable!()
```
After:
```rust
        Referrer::Client =>
        // it should be impossible for referrer to be anything else during fetching
        // https://fetch.spec.whatwg.org/#concept-request-referrer
        {
            unreachable!()
        },
```

### PR DESCRIPTION
**Pull request by fyrchik**
This fixes #3170 

This looks not so elegant, however I have not found a simpler way, because there are at least 4 different cases, which can be found in test.

Let me know, what you think.

### CODE REVIEW

**Comment by scampi** [comment]
@topecongiro should this be version-gated ?

File: src/matches.rs [hunk 1]
```diff
@@ -430,19 +439,26 @@ fn rewrite_match_body(
...
+        // the arm body block.
         let mut result = pats_str.to_owned();
         result.push_str(" =>");
-        if !arrow_comment.is_empty() {
+        if !arrow_comment.is_empty() && was_block {
             result.push_str(&indent_str);
             result.push_str(&arrow_comment);
         }
         result.push_str(&block_sep);
+        if !arrow_comment.is_empty() && !was_block {
```
[the remark this answers is not in the document]
Comment (fyrchik) [reply src/matches.rs]: If our match arm wasn't a block and had an arrow comment, then this comment must be inside the block.
This is case 1 in tests and the case described in the issue.

File: src/matches.rs [hunk 2]
```diff
@@ -386,14 +392,17 @@ fn rewrite_match_body(
         let comment_str = arrow_snippet[arrow_index + 2..].trim();
         if comment_str.is_empty() {
             String::new()
+        } else if (!was_block && !is_block) || is_empty_block {
```
Comment (scampi) [inline src/matches.rs]: Which case in the test of the issue is covered by `!was_block && !is_block` ?

**Comment by topecongiro** [comment]
> @topecongiro I think this change should be version-gated ?

Yes.

File: src/matches.rs (hunk 2 above)
Comment (fyrchik) [reply src/matches.rs]: It is case 1, when we have single statement match arm with arrow-comment.

Comment (scampi) [reply src/matches.rs]: Thanks

**Pull request merged**